src/cli/self_update/types.rs

```diff
@@ -0,0 +1,8 @@
+#![allow(dead_code)]
+use std::path::PathBuf;
```

Comment:
Do you know why this triggers  a dead_code lint?

I'd rather an affirmative name for this file rather than generic 'types'; it seems like just trading one big bucket for a small bucket will grow otherwise.

e.g. 'pathupdate.rs'.

Reply:
1) I really don't know, I will make another go at removing the dead_code, but this was a weirdly "entangled" type so that might be why?
2) Completely reasonable!

I am working on the code that will make that type fully redundant, which is why I was indifferent to this file being sloppy.

Reply:
Oh, I realize why now. Splitting out the method which generates a Vec of that type warns "variant never constructed". Still, renaming the file is easy enough.

Reply:
@workingjubilee Easy enough if you actually follow up and put the change in all the places that require it, Past!Jubilee.